### PR TITLE
docs: Revise spec contents on types (2/4)

### PR DESCRIPTION
**Pull request by juliaogris**
Revise spec contents on types and use evy and evy:output code fences.
The context of these changes is as follows although there have been
minor reformatting changes added to the later sections too.

✅ Intro
✅ Evy syntax specification
👉 Comments
👉 Types
👉 Variables and Declarations
👉 Zero Values
👉 Assignments
👉 Copy and reference
👉 Scope
👉 Strings
👉 Arrays
👉 Maps
👉 Index and Slice
⬜️ Operators
⬜️ Precedence
⬜️ Whitespace
⬜️ Functions
⬜️ Variadic functions
⬜️ Break and Return
⬜️ Typeof
⬜️ Type assertion
⬜️ Event Handler
⬜️ Errors
⬜️ Runtime

Add doctest for spec.md to Makefile and CI.

### CODE REVIEW

File: docs/spec.md [hunk 1]
```diff
@@ -265,252 +268,441 @@ according to the IEEE-754 64-bit floating point standard.
...
 
-    s := "letters"
-    print m[s]         // abc
+```evy:output
+1 true
+2 false
+```
 
-The `has` function tests for the existence of a key in a map:
+The `del` function deletes a key from a map if it exists. The following
```
Comment (camh-) [inline docs/spec.md:620]: ...exists and does nothing if the key does not exist.

**Review by camh- — APPROVED** [review]
:green_book: Excellent! Reads very well. Just one very trivial comment.

File: docs/spec.md [hunk 2]
```diff
@@ -204,252 +268,441 @@ according to the IEEE-754 64-bit floating point standard.
...
+inferred declarations and typed declarations.
 
-Variables must be _declared_ before they can be used. A variable
-declaration can either be an _inferred declaration_ or a _typed
-declaration_. With the inferred declaration the type is not given but
-inferred from the value. With typed declaration the type is explicitly
-specified and the variable is initialised to its type's zero value.
+_Inferred declarations_ do not specify the type of the variable
+explicitly. The type of the variable is inferred from the value that
+it is initialised to. For example, the following code declares a
```
Comment (camh-) [inline docs/spec.md:277]: s/initialised/initialized/

File: docs/spec.md [hunk 3]
```diff
@@ -204,252 +268,441 @@ according to the IEEE-754 64-bit floating point standard.
...
+variable `n` and initialises it to the value `1`. The type of `n` is
```
Comment (camh-) [inline docs/spec.md:278]: s/initialised/initialized/

File: docs/spec.md [hunk 4]
```diff
@@ -204,252 +268,441 @@ according to the IEEE-754 64-bit floating point standard.
...
+arr5 := [1] + [] // type: []num
+
+map1 := {} // type: {}any
+map2 := {age:10} // type: {}num
+print arr1 arr2 arr3 arr4 arr5 map1 map2
+```
 
 ## Zero Values
 
 Variables declared via typed declaration are initialised to the zero
```
Comment (camh-) [inline docs/spec.md:308]: s/initialised/initialized/

**Review by camh- — APPROVED** [review]

File: docs/builtins.md [hunk 5]
```diff
@@ -426,7 +426,8 @@ Output
 The `del` function takes two arguments: a map and a key. It deletes the
 key-value entry from the map if the key exists. If the key does not
 exist, the function does nothing. The map can have any value type, and
-the key can be any string.
+the key can be any string. It is safe to delete values from the map
+with `del` while iterating.
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply docs/builtins.md:430]: yes, much better, done.

File: docs/spec.md [hunk 6]
```diff
@@ -265,252 +268,441 @@ according to the IEEE-754 64-bit floating point standard.
...
+initializes it to the value `"a"` through inference. Then, it assigns
+the value `"b"` to `s`. Finally, it tries to assign the value `100` to
+`s`, which will cause a compile-time error because `s` is a string
+variable and `100` is a number.
+
+```evy
+s := "a"
+print 1 s
+s = "b"
+print 2 s // prints 2
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply docs/spec.md:360]: yes, sorry, removed.

File: docs/spec.md (hunk 1 above)
Comment (juliaogris) [reply docs/spec.md:620]: yes!

File: docs/spec.md (hunk 4 above)
Comment (juliaogris) [reply docs/spec.md:308]: done.

File: docs/spec.md (hunk 3 above)
Comment (juliaogris) [reply docs/spec.md:278]: done.

**Review by camh- — APPROVED** [review]
:green_book:

**Pull request merged**